purchase_add_product_supplierinfo/purchase.py

```diff
@@ -0,0 +1,53 @@
+# -*- coding: utf-8 -*-
+# © 2016 Chafique DELLI @ Akretion
+# License AGPL-3.0 or later (http://www.gnu.org/licenses/agpl.html).
+
+from openerp import models, api, _
+
+
+class PurchaseOrder(models.Model):
+    _inherit = 'purchase.order'
+
+    @api.model
+    def _check_product_supplierinfo(self):
+        lines = []
+        for line in self.order_line:
+            suppinfo = self.env['product.supplierinfo'].search([
+                ('product_id', '=', line.product_id.id),
+                ('name', '=', self.partner_id.id)])
```

Comment:
Wrong domain here.
If you have a generic supplier for all of the variant the product_id field of the supplier will be empty